src/Interpreters/InterpreterKillQueryQuery.cpp

```diff
@@ -277,9 +277,11 @@ BlockIO InterpreterKillQueryQuery::execute()
                     code = CancellationCode::NotFound;
                 else
                 {
-                    ParserAlterCommand parser;
+                    const auto alter_command = command_col.getDataAt(i).toString();
+                    const auto with_round_bracket = alter_command.front() == '(';
+                    ParserAlterCommand parser{with_round_bracket};
```

Comment:
Does it make sense to always allow round brackets when parsing, despite the value of the setting? So the setting will be only used for formatting, and the parser will be able to always understand both versions

Reply:
The idea is by allowing parsing with parentheses by default is the transition can be smoother during migrating to newer versions. E.g. during live migration if there are one replica which produces alter commands with parens, then the older replicas will parse it properly.

Reply:
Maybe my previous answer wasn't the best. So I wanted to avoid allowing the "mix" of the old and new syntax. So in an alter command list each command must or mustn't have to be surrounded with parentheses. E.g.: `ALTER TABLE x MODIFY TTL, DROP COLUM a` is fine, `ALTER TABLE x (MODIFY TTL), (DROP COLUMN a)` also fine, but `ALTER TABLE x (MODIFY TTL), DROP COLUMN a` is not fine.

This means in some way `ParserAlterCommandList` has to be able to enforce it. This could be done in two ways in my opinion:
1. The list parser figures this out and passes it to the individual command parser.
2. The individual command parser decides whether parens should be used or not and because in `ParserAlterCommandList::parseImpl` only a single parser is used, it can carry over this information. It is doable, but it would introduce in my opnion a hidden relation between the first parsed command and every command after that. I think right now parsers doesn't have this kind of "memory" in general. Thus, I opted for option 1.

That's why here determining `with_round_bracket` is necessary. What do you think?

Reply:
> So in an alter command list each command must or mustn't have to be surrounded with parentheses.

Yes, I agree.

> This means in some way ParserAlterCommandList has to be able to enforce it. ... That's why here determining with_round_bracket is necessary. 

Sorry, I missed this when reading the code, it was unobvious.
Yes, the first option is better because in the worst case, someone will pass `true` or `false` unconditionally instead of proper `with_round_bracket` flag (so only one version of the syntax will work in this place). And with the second option, someone may use one `ParserAlterCommand` for multiple queries, and it will cause an obscure bug